File: SDKs/envoice.in/v1/python/src/sdk/models/shared/invoice.py
from dataclasses import dataclass, field
from datetime import datetime
from marshmallow import fields
import dateutil.parser
from typing import Enum,List,Optional
from dataclasses_json import dataclass_json
from . import activity
from . import invoiceattachment
from . import invoiceitem
from . import paymentgatewayforinvoice
from . import payment

class InvoiceStatusEnum(str, Enum):
    DRAFT = "Draft"
    PAID = "Paid"
    UNPAID = "Unpaid"
    OVERDUE = "Overdue"
    VOID = "Void"


@dataclass_json
@dataclass
class Invoice:
    access_token: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'AccessToken' }})
    activities: Optional[List[activity.Activity]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Activities' }})
    attachments: Optional[List[invoiceattachment.InvoiceAttachment]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Attachments' }})
    client_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'ClientId' }})
    cloned_from_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'ClonedFromId' }})
    currency_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'CurrencyId' }})
    discount_amount: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'DiscountAmount' }})
    duedate: Optional[datetime] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Duedate', 'encoder': datetime.isoformat, 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    enable_partial_payments: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'EnablePartialPayments' }})
    estimation_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'EstimationId' }})
    id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Id' }})
    invoice_category_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'InvoiceCategoryId' }})
    issued_on: Optional[datetime] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'IssuedOn', 'encoder': datetime.isoformat, 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    items: Optional[List[invoiceitem.InvoiceItem]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Items' }})
    notes: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Notes' }})
    number: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Number' }})
    order_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'OrderId' }})
    payment_gateways: Optional[List[paymentgatewayforinvoice.PaymentGatewayForInvoice]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'PaymentGateways' }})
    payment_link_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'PaymentLinkId' }})
    payments: Optional[List[payment.Payment]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Payments' }})
    po_number: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'PoNumber' }})
    recurring_profile_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'RecurringProfileId' }})
    should_send_reminders: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'ShouldSendReminders' }})
    status: Optional[InvoiceStatusEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Status' }})
    sub_total_amount: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'SubTotalAmount' }})
    tax_amount: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'TaxAmount' }})
    terms: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Terms' }})
    total_amount: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'TotalAmount' }})
    user_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'UserId' }})
    

File: SDKs/envoice.in/v1/typescript/src/sdk/models/shared/invoice.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Activity } from "./activity";
import { InvoiceAttachment } from "./invoiceattachment";
import { InvoiceItem } from "./invoiceitem";
import { PaymentGatewayForInvoice } from "./paymentgatewayforinvoice";
import { Payment } from "./payment";


export enum InvoiceStatusEnum {
    Draft = "Draft",
    Paid = "Paid",
    Unpaid = "Unpaid",
    Overdue = "Overdue",
    Void = "Void"
}


export class Invoice extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=AccessToken" })
  accessToken?: string;

  @SpeakeasyMetadata({ data: "json, name=Activities", elemType: Activity })
  activities?: Activity[];

  @SpeakeasyMetadata({ data: "json, name=Attachments", elemType: InvoiceAttachment })
  attachments?: InvoiceAttachment[];

  @SpeakeasyMetadata({ data: "json, name=ClientId" })
  clientId?: number;

  @SpeakeasyMetadata({ data: "json, name=ClonedFromId" })
  clonedFromId?: number;

  @SpeakeasyMetadata({ data: "json, name=CurrencyId" })
  currencyId?: number;

  @SpeakeasyMetadata({ data: "json, name=DiscountAmount" })
  discountAmount?: number;

  @SpeakeasyMetadata({ data: "json, name=Duedate" })
  duedate?: Date;

  @SpeakeasyMetadata({ data: "json, name=EnablePartialPayments" })
  enablePartialPayments?: boolean;

  @SpeakeasyMetadata({ data: "json, name=EstimationId" })
  estimationId?: number;

  @SpeakeasyMetadata({ data: "json, name=Id" })
  id?: number;

  @SpeakeasyMetadata({ data: "json, name=InvoiceCategoryId" })
  invoiceCategoryId?: number;

  @SpeakeasyMetadata({ data: "json, name=IssuedOn" })
  issuedOn?: Date;

  @SpeakeasyMetadata({ data: "json, name=Items", elemType: InvoiceItem })
  items?: InvoiceItem[];

  @SpeakeasyMetadata({ data: "json, name=Notes" })
  notes?: string;

  @SpeakeasyMetadata({ data: "json, name=Number" })
  number?: string;

  @SpeakeasyMetadata({ data: "json, name=OrderId" })
  orderId?: number;

  @SpeakeasyMetadata({ data: "json, name=PaymentGateways", elemType: PaymentGatewayForInvoice })
  paymentGateways?: PaymentGatewayForInvoice[];

  @SpeakeasyMetadata({ data: "json, name=PaymentLinkId" })
  paymentLinkId?: number;

  @SpeakeasyMetadata({ data: "json, name=Payments", elemType: Payment })
  payments?: Payment[];

  @SpeakeasyMetadata({ data: "json, name=PoNumber" })
  poNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=RecurringProfileId" })
  recurringProfileId?: number;

  @SpeakeasyMetadata({ data: "json, name=ShouldSendReminders" })
  shouldSendReminders?: boolean;

  @SpeakeasyMetadata({ data: "json, name=Status" })
  status?: InvoiceStatusEnum;

  @SpeakeasyMetadata({ data: "json, name=SubTotalAmount" })
  subTotalAmount?: number;

  @SpeakeasyMetadata({ data: "json, name=TaxAmount" })
  taxAmount?: number;

  @SpeakeasyMetadata({ data: "json, name=Terms" })
  terms?: string;

  @SpeakeasyMetadata({ data: "json, name=TotalAmount" })
  totalAmount?: number;

  @SpeakeasyMetadata({ data: "json, name=UserId" })
  userId?: number;
}
